lib/foreman_inventory_upload/generators/fact_helpers.rb

```diff
@@ -114,7 +114,7 @@ def hostname_match
         foreman_hostname = ForemanRhCloud.foreman_host&.name
         if bash_hostname == foreman_hostname
           fqdn(ForemanRhCloud.foreman_host)
-        elsif Setting[:obfuscate_inventory_hostnames]
+        elsif CommonParameter.find_by(name: 'obfuscate_inventory_hostnames')&.value
```

Comment:
If it's a parameter, you have to look up the value on the host object itself: https://github.com/theforeman/foreman_rh_cloud/blob/48f9d74ba8a671c74cec3a79b2646c38172a42a3/app/controllers/insights_cloud/api/machine_telemetries_controller.rb#L87